src/Security/Helper/SaveAuthFailureMessage.php

```diff
@@ -22,7 +23,11 @@ protected function saveAuthenticationErrorToSession(Request $request, Authentica
         if (!$request->hasSession() || !$request->getSession() instanceof SessionInterface) {
             throw new \LogicException('In order to save an authentication error, you must have a session available.');
         }
+        
+        $securityClass = class_exists(SecurityRequestAttributes::class)
+            ? SecurityRequestAttributes::class
+            : Security::class;
 
-        $request->getSession()->set(Security::AUTHENTICATION_ERROR, $exception);
+        $request->getSession()->set($securityClass::AUTHENTICATION_ERROR, $exception);
```

Comment:
Hm, but `$securityClass` is a simple string right? I'm afraid we will have an error here. We need to set the error on a var above and use the var here

Reply:
it dosnet matter, its right this, maybe older php not 🤷 oki, i am goint to change it,  it will work, but its not pretty

Reply:
@bocharsky-bw its probably readable and more explicit, see agaign, thx 🙂

Reply:
i see pipeline failed, but its not related with this change 🤷